casper/src/main/resources/PoS.rhox

```diff
@@ -344,17 +411,25 @@ new  PoS,
           }
         } |
 
-        contract deposit (@fromUserPk, @amount, returnCh) = {
+        contract deposit (@fromUserPk, @amount, @activeValidators, returnCh) = {
           new vaultCh,
               revAddressCh,
+              indexSender,
+              getBlockData(`rho:block:data`), blockDataCh,
               authKeyCh in {
             @RevVault!("deployerAuthKey", *authKeyCh) |
+            getBlockData!(*blockDataCh) |
             for (@authKey <- authKeyCh) {
               revAddressOps!("fromPublicKey", fromUserPk, *revAddressCh) |
               for (@fromRevAddress <- revAddressCh) {
                 @RevVault!("findOrCreate", fromRevAddress, *vaultCh) |
-                for (@(true, fromVault) <- vaultCh) {
-                  @fromVault!("transfer", posRevAddress, amount, authKey, *returnCh)
+                for (@(true, fromVault) <- vaultCh; _, _, @sender <- blockDataCh) {
```

Comment:
Joined channels are more prone to cause conflicts because the mergeability rules are not applied to them. 
I'm not sure if it makes sense to get rid of this particular join, just double-checking.

Reply:
Hmm, why "more prone"? In which cases are they not conflicting?

Reply:
When disparate channels are used in both blocks